subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv_aux.c

```diff
@@ -477,9 +543,102 @@ static inline int isr_rx_pdu(struct lll_adv_aux *lll_aux,
 					 CONFIG_BT_CTLR_GPIO_PA_OFFSET);
 #endif /* CONFIG_BT_CTLR_GPIO_PA_PIN */
 		return 0;
-	}
+	} else if ((pdu_rx->type == PDU_ADV_TYPE_AUX_CONNECT_REQ) &&
+		   (pdu_rx->len == sizeof(struct pdu_adv_connect_ind)) &&
+		   lll_adv_connect_ind_check(lll, pdu_rx, tx_addr, addr,
+					     rx_addr, tgt_addr,
+					     devmatch_ok, &rl_idx)) {
+		struct node_rx_ftr *ftr;
+		struct node_rx_pdu *rx;
+
+		if (IS_ENABLED(CONFIG_BT_CTLR_CHAN_SEL_2)) {
+			rx = ull_pdu_rx_alloc_peek(4);
+		} else {
+			rx = ull_pdu_rx_alloc_peek(3);
+		}
+
+		if (!rx) {
+			return -ENOBUFS;
+		}
 
-	/* TODO: add handling for AUX_CONNECT_REQ */
+		radio_isr_set(isr_tx_connect_rsp, rx);
+		radio_switch_complete_and_disable();
+		radio_pkt_tx_set(prepare_connect_rsp_pdu(pdu_rx));
+
+		/* assert if radio packet ptr is not set and radio started tx */
+		LL_ASSERT(!radio_is_ready());
+
+		if (IS_ENABLED(CONFIG_BT_CTLR_PROFILE_ISR)) {
+			lll_prof_cputime_capture();
+		}
+
+#if defined(CONFIG_BT_CTLR_GPIO_PA_PIN)
+		if (IS_ENABLED(CONFIG_BT_CTLR_PROFILE_ISR)) {
+			/* PA/LNA enable is overwriting packet end used in ISR
+			 * profiling, hence back it up for later use.
+			 */
+			lll_prof_radio_end_backup();
+		}
+
+		radio_gpio_pa_setup();
+		radio_gpio_pa_lna_enable(radio_tmr_tifs_base_get() +
+					 EVENT_IFS_US -
+					 radio_rx_chain_delay_get(0, 0) -
+					 CONFIG_BT_CTLR_GPIO_PA_OFFSET);
+#endif /* CONFIG_BT_CTLR_GPIO_PA_PIN */
+
+		/* Note: this is the same as previous result from alloc_peek */
+		rx = ull_pdu_rx_alloc();
+
+		rx->hdr.type = NODE_RX_TYPE_CONNECTION;
+		rx->hdr.handle = 0xffff;
+
+		memcpy(rx->pdu, pdu_rx, (offsetof(struct pdu_adv, connect_ind) +
+					 sizeof(struct pdu_adv_connect_ind)));
+		ftr = &(rx->hdr.rx_ftr);
+		ftr->param = lll;
+		ftr->ticks_anchor = radio_tmr_start_get();
+		ftr->radio_end_us = radio_tmr_end_get() -
+				    radio_tx_chain_delay_get(0, 0);
+
+#if defined(CONFIG_BT_CTLR_PRIVACY)
+		ftr->rl_idx = irkmatch_ok ? rl_idx : FILTER_IDX_NONE;
+#endif /* CONFIG_BT_CTLR_PRIVACY */
+
+		if (IS_ENABLED(CONFIG_BT_CTLR_CHAN_SEL_2)) {
+			ftr->extra = ull_pdu_rx_alloc();
+		}
+
+		return 0;
+	}
 
 	return -EINVAL;
 }
+
+static void isr_tx_connect_rsp(void *param)
+{
+	struct node_rx_ftr *ftr;
+	struct node_rx_pdu *rx;
+	struct lll_adv *lll;
+	int ret;
+
+	rx = param;
+	ftr = &(rx->hdr.rx_ftr);
+	lll = ftr->param;
+
+	if (!radio_is_done()) {
+		/* TODO: release node_rx somehow */
```

Comment:
I am not sure how to do this properly: free it here somehow (did not see any code that does similar thing) or send and handle as error in ull?

Basically node_rx is allocated when AUX_CONNECT_REQ is received because we at that point in time we need to decide whether we have resources to accept connection and also that PDU is used as an anchor point. However, we should not send node until AUX_CONNECT_RSP is sent as otherwise we may end with a connection that was not properly established.